### PR TITLE
Add support for testing with CSP version 1

### PR DESCRIPTION
**Pull request by takumiando**
- Add CSP version option to all the examples
- Add tests for CSP version 1

This patch set is to avoid bugs specific to CSP version 1.

### CODE REVIEW

**Comment by takumiando** [comment]
This is related to https://github.com/libcsp/libcsp/issues/708.

**Comment by yashi** [comment]
Work on the commit message a bit.  "Modify for CSP version 1" doesn't tell you anything.  Any commit is a modification.

**Comment by takumiando** [comment]
> Work on the commit message a bit. "Modify for CSP version 1" doesn't tell you anything. Any commit is a modification.

I changed PR title.

Currently, the added tests for CSP v1 fail because of the issue https://github.com/libcsp/libcsp/issues/708.

File: .github/workflows/build-test.yml
Comment (yashi) [inline .github/workflows/build-test.yml:1]: Can we do this with [github actions matrix](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow)? Having copies of everything is tedious.

Comment (takumiando) [reply .github/workflows/build-test.yml:1]: I'll try it.

**Comment by takumiando** [comment]
like this?
image

**Comment by yashi** [comment]
Thanks. That's more manageable. LGTM but this need to wait for a bit to be merged.

**Comment by yashi** [comment]
@takumiando Please rebase it.

**Pull request merged**